src/scheduler/resv_info.cpp

```diff
@@ -1698,20 +1658,26 @@ confirm_reservation(status *policy, int pbs_sd, resource_resv *unconf_resv, serv
 		 * each execvnode and condensing the concatenated string.
 		 */
 		if (nresv_parent->resv->is_standing)
-			short_xc = condense_execvnode_seq(execvnodes);
+			short_xc = condense_execvnode_seq(execvnodes.c_str());
 		else
-			short_xc = string_dup(execvnodes);
+			short_xc = string_dup(execvnodes.c_str());
+		
+		if (short_xc == NULL || get_execvnodes_count(short_xc) != occr_count) {
```

Comment:
If get_execvnodes_count() is only needed to make sure that condense_execvnode_seq() didn't screw up, maybe just have condense_execvnode_seq() check that the exec vnode count matches occr_count and return NULL if it doesn't? otherwise we will incur the overhead of get_execvnodes_count() for advance reservations as well. I think we can even make direct_map() return the count instead of calling get_execvnodes_count() for standing reservations.

Reply:
condense_execvnode_seq() works off of a string.  You don't pass in a reservation.  We'd have to pass in the occr_count into the function.  I don't think condense_execvnode_seq() is the right place for this check.  I think it the right place is here.

Reply:
condense_execvnode_seq() is condensing an exec vnode, so it seems well within its purview to ensure that it kept the number of occurrences the same while condensing it. If you still think that it doesn't belong there, then please move the get_execvnodes_count(short_xc) != occr_count check to the if (nresv_parent->resv->is_standing) block so that it doesn't get performed for advance reservations